### PR TITLE
Add Platform Validation Framework document

### PR DESCRIPTION
**Pull request by jiere**
This is document for Kepler's PR [#902](https://github.com/sustainable-computing-io/kepler/pull/902)

### CODE REVIEW

File: docs/platform-validation/index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+Platform validation cases should support both validity and accuracy check on Kepler/Prometheus exposed data.
+
+Take Intel X86 BareMetal platform's validation as example, we have introduced an independent RAPL-based energy collection and power consumption calculation tool called `validator`.
+
+The current work mechanism and features of `validator` are simple:
+1. It could detect the current platform's CPU model type.
+2. It could detect the current platform's inband RAPL components support status and OOB platform power source support status.
+3. It uses specific sampling count(configurable, by default 20 sampling cycles) with specific sampling interval(configurable, by default 15 seconds) to collect the specific components' RAPL values and calculate the power consumption of each sampling, then achieve the mean value of node components power among those sampling cycles.
+
+Test cases could use above sampling and calculation results as comparison base to check the Kepler exported and Promethes aggregated query results.
```
Comment (rootfs) [inline docs/platform-validation/index.md:24]: `Promethes` -> `Prometheus`

File: docs/platform-validation/index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+### Artifacts build.
+
+Build both the `Kepler` container image and the `Validator` container image.
+
+The current Validator image refers to the Kepler image build mechanism to build from a GPU image to support GPU case. We may simply build it from UBI image later.
+
+Since it is an independent energy collector other than Kepler, those eBPF stuffs will not be bumped into the container.
+
+### Platform validation test.
```
Comment (rootfs) [inline docs/platform-validation/index.md:92]: TODO: please get a script for these steps.

File: docs/platform-validation/index.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+8. Cleanup the test environment(undeploy Kepler exporter, the local-dev-cluster and the local registry).
+
+
+## Validation result evaluation.
+
+Check validation report for any failure cases.
+
+Check accuracy cases comparison result. 
+
+Node level cases should be more accurate than pod/container level cases, make analysis on any abnormal results and figure out RCAs for them.
```
Comment (rootfs) [inline docs/platform-validation/index.md:127]: TODO create an analyzer for abnormality detection

File: docs/platform-validation/index.md (hunk 2 above)
Comment (jiere) [reply docs/platform-validation/index.md:92]: Currently these steps have been automated in Github action workflow, see Kepler PR#902.
.gitub/workflows/platform-validation.yml and hack/verify.sh.

File: docs/platform-validation/index.md (hunk 3 above)
Comment (jiere) [reply docs/platform-validation/index.md:127]: Yeah, this is really TODO item which I am doing, will update when it is ready.

File: docs/platform-validation/index.md (hunk 1 above)
Comment (jiere) [reply docs/platform-validation/index.md:24]: sorry for typo, fix soon.

**Review by rootfs — APPROVED** [review]

**Pull request merged**